app/components/common/navbar/index.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useState, useEffect, useRef } from 'react';
+import {
+  HomeIcon,
+  CalendarIcon,
+  UsersIcon,
+  Cog6ToothIcon,
+  QuestionMarkCircleIcon,
+} from '@heroicons/react/24/outline';
+
+const Navbar = () => {
+  const [active, setActive] = useState<number>(0);
+  const [expand, setExpand] = useState<boolean>(false);
+  const navbarRef = useRef<HTMLDivElement>(null!);
+
+  const myColor = (elementPosition: number) => {
+    if (active === elementPosition) return true;
+    return false;
+  };
+  const toggle = (elementPosition: number) => {
+    if (active !== elementPosition) setActive(elementPosition);
+  };
+  useEffect(() => {
+    const observer = new window.ResizeObserver((entries) => {
+      const navbarElementWidth = entries[0].borderBoxSize[0].inlineSize;
+      const navbarElementHeight = entries[0].borderBoxSize[0].blockSize;
+
+      if (navbarElementHeight <= 110 && navbarElementWidth < 768) {
+        if (!expand) {
+          if (!expand) setExpand(false);
+        }
+      }
+    });
+    observer.observe(navbarRef.current);
```

Comment:
Can this not be done using CSS media queries instead?

Reply:
i don't think we can update state  based of media queries, I tried finding something similar but i couldn't find anything which would help me here. You can share a resource if you have any

Reply:
From what I was able to understand we added a resize observer and depending on the viewport width navbar height & width are being set. I am assuming this is to make the navbar responsive in mobile view. 

If my intuition is correct the navbar can be made responsive using CSS media queries where instead of checking the viewport width using javascript you can define a media query to check the same and set its width and height accordingly. 

Here is a link to a tailwind doc that can help: [https://tailwindcss.com/docs/responsive-design](https://tailwindcss.com/docs/responsive-design)

Reply:
I think you got it wrong, We are using a state variable to expand the navbar in the web version, but since we don't want the expanded version of the navbar in the mobile view, we are changing the state based on the size of the navbar width and height. 
For now I will make the changes as discussed in the call.